src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -324,6 +328,33 @@ def post(self, emulator=None):
 
         return None
 
+    @staticmethod
+    def getSingleScramArch(scramArch):
+        """
+        Figure out which scram arch is compatible with both the request and the release
+
+        Args:
+            scramArch: string or list of strings representing valid scram arches for the workflow
+
+        Returns:
+            a single scram arch
+        """
+
+        if isinstance(scramArch, list):
+            try:
+                ad = readAdValues(['glidein_required_os'], 'machine')
+                runningOS = ad['glidein_required_os'].strip('"')
+                validArches = sorted(OS_TO_ARCH[runningOS], reverse=True)
+                for requestedArch in sorted(scramArch, reverse=True):
+                    for validArch in validArches:
+                        if requestedArch.startswith(validArch):
+                            return requestedArch
+            except KeyError:
+                return sorted(scramArch)[-1]  # Give the most recent release if lookup fails
```

Comment:
It may be worth to log this error, with runningOS and scramArch.
In addition to that, I'd log the case we have no match (return None).